pkg/token/token.go

```diff
@@ -410,6 +422,14 @@ func stsHostsForPartition(partitionID, region string) map[string]bool {
 	stsSvc, ok := partition.Services()[stsServiceID]
 	if !ok {
 		logrus.Errorf("STS service not found in partition %s", partitionID)
+		// Add the host of the current instances region if the service doesn't already exists in the partition
+		// so we don't fail if the service is not present in the go sdk but matches the instances region.
+		stsHostName, err := getDefaultHostNameForRegion(partition, region)
+		if err != nil {
+			logrus.WithError(err).Error("Error getting default hostname")
+		} else {
+			validSTShostnames[stsHostName] = true
```

Comment:
qs for familiarity: can you describe how this helps? when the sdk does not have the partition, the else block will not be taken. How does the rest of the flow succeed?

Reply:
when the sdk doesn't have partition we fail as we are now aware of what a default dns hostname should be. This would help where there is partition but the services are not yet defined under the partition.